presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java

```diff
@@ -0,0 +1,161 @@
+
+            checkArgument(isSupportedType(columnHandle.getType()), "Unsupported column type '%s' for column '%s'", columnHandle.getType(), columnHandle.getName());
+        }
+        this.byteArrayOuts = new ByteArrayOutputStream();
+        this.parsedSchema = requireNonNull(parsedSchema, "parsedSchema is null");
+        this.dataFileWriter = new DataFileWriter<>(new GenericDatumWriter<>(this.parsedSchema));
+        this.record = new GenericData.Record(this.parsedSchema);
+    }
+
+    private boolean isSupportedType(Type type)
+    {
+        return isVarcharType(type) || SUPPORTED_PRIMITIVE_TYPES.contains(type);
+    }
+
+    @Override
+    protected void appendNullValue()
+    {
+        record.put(columnHandles.get(currentColumnIndex).getName(), null);
+    }
+
+    @Override
+    protected void appendLong(long value)
+    {
+        record.put(columnHandles.get(currentColumnIndex).getName(), value);
+    }
+
+    @Override
+    protected void appendInt(int value)
+    {
+        record.put(columnHandles.get(currentColumnIndex).getName(), value);
+    }
+
+    @Override
+    protected void appendShort(short value)
+    {
+        record.put(columnHandles.get(currentColumnIndex).getName(), value);
+    }
+
+    @Override
+    protected void appendByte(byte value)
+    {
+        record.put(columnHandles.get(currentColumnIndex).getName(), value);
+    }
+
+    @Override
+    protected void appendDouble(double value)
+    {
+        record.put(columnHandles.get(currentColumnIndex).getName(), value);
+    }
+
+    @Override
+    protected void appendFloat(float value)
+    {
+        record.put(columnHandles.get(currentColumnIndex).getName(), value);
+    }
+
+    @Override
+    protected void appendBoolean(boolean value)
+    {
+        record.put(columnHandles.get(currentColumnIndex).getName(), value);
+    }
+
+    @Override
+    protected void appendString(String value)
+    {
+        record.put(columnHandles.get(currentColumnIndex).getName(), value);
+    }
+
+    @Override
+    public byte[] toByteArray()
+    {
+        // make sure entire row has been updated with new values
+        checkArgument(currentColumnIndex == columnHandles.size(), format("Missing %d columns", columnHandles.size() - currentColumnIndex + 1));
+
+        try {
+            byteArrayOuts.reset();
+            dataFileWriter.create(parsedSchema, byteArrayOuts);
+            dataFileWriter.append(record);
+            dataFileWriter.flush();
+
+            resetColumnIndex(); // reset currentColumnIndex to prepare for next row
+            return byteArrayOuts.toByteArray();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException("Failed to append record", e);
+        }
+    }
+
+    @Override
+    public void close()
+    {
+        try {
+            byteArrayOuts.close();
+            dataFileWriter.close();
```

Comment:
Nothing wrong with this, but it didn't seem necessary.  `ByteArrayOutputStream.close()` is really a no-op and I don't think the Avro writer holds on to any resources.